docs/architecture/adr-035-rosetta-api-support.md

```diff
@@ -0,0 +1,225 @@
+# ADR 035: Rosetta API Support
+
+## Context
+
+[Rosetta API](https://www.rosetta-api.org/) is an open-source specification and set of tools developed by Coinbase to standardise blockchain interactions.
+
+Through the use of a standard API for integrating blockchain applications it will
+
+* Be easier for a user to interact with a given blockchain
+* Allow exchanges to integrate new blockchains quickly and easily
+* Enable application developers to build cross-blockchain applications such as block explorers, wallets and dApps at considerably lower cost and effort.
+
+## Decision
+
+It is clear that adding Rosetta API support to the Cosmos SDK will bring value to all the developers and Cosmos SDK based chains in the ecosystem. How it is implemented is key.
+
+The driving principles of the proposed design are:
+
+1. **Extensibility:** it must be as riskless and painless as possible for application developers to set-up network configurations to expose Rosetta API-compliant services.
+2. **Long term support:** This proposal aims to provide support for all the supported Cosmos SDK release series.
```

Comment:
What's a release series? It's not clear from this statement which exact versions you plan on backporting this functionality to.

Reply:
0.40.1 -> release series is `0.40`
0.40.2 -> release series is `0.40`
0.39.1 -> release series is `0.39`

[so on and so forth]

Reply:
And how far back?

Reply:
We originally planned to deliver support for historical data as well (meaning: 0.37). As you may imagine, that's low priority at the moment though.